Framework/DataHandling/src/GenerateGroupingPowder.cpp

```diff
@@ -73,29 +74,26 @@ void GenerateGroupingPowder::init() {
  */
 void GenerateGroupingPowder::exec() {
   MatrixWorkspace_const_sptr input_ws = getProperty("InputWorkspace");
-  // check if workspace has an instrument. If not, throw an exception
-  Mantid::Geometry::Instrument_const_sptr instrument =
-      input_ws->getInstrument();
-  std::vector<detid_t> dets = instrument->getDetectorIDs(true);
-  if (dets.empty()) {
+  // Check if workspace has detectors. If not, throw an exception.
+  const auto &detectorInfo = input_ws->detectorInfo();
+  auto dets = detectorInfo.detectorIDs();
```

Comment:
These are probably better labelled as detector ids i.e. `detIDs`. Wouldn't it be better to store a reference rather than a copy of these?